### PR TITLE
Add new documentation grammar in Ballerina IDEA plugin

### PR DESCRIPTION
**Pull request by Shan1024**
## Purpose
Add new documentation grammar support in Ballerina IDEA plugin.

selection_390

### CODE REVIEW

File: tool-plugins/intellij/src/main/java/org/ballerinalang/plugins/idea/psi/references/DocVariableReference.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (NipunaMarcus) [inline tool-plugins/intellij/src/main/java/org/ballerinalang/plugins/idea/psi/references/DocVariableReference.java]: year is wrong, should be 2018

File: tool-plugins/intellij/src/main/java/org/ballerinalang/plugins/idea/psi/references/DocVariableReference.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            }
+            PsiElement nextSibling = PsiTreeUtil.skipSiblingsForward(documentationAttachmentNode, PsiWhiteSpace.class,
+                    PsiComment.class);
+            if (nextSibling == null) {
+                return null;
+            }
+            if (nextSibling instanceof DefinitionNode) {
+
+            }
+
```
Comment (NipunaMarcus) [inline tool-plugins/intellij/src/main/java/org/ballerinalang/plugins/idea/psi/references/DocVariableReference.java]: unnecessary spaces

File: tool-plugins/intellij/src/main/java/org/ballerinalang/plugins/idea/psi/references/DocVariableReference.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            if (documentationAttachmentNode == null) {
+                return null;
+            }
+            PsiElement nextSibling = PsiTreeUtil.skipSiblingsForward(documentationAttachmentNode, PsiWhiteSpace.class,
+                    PsiComment.class);
+            if (nextSibling == null) {
+                return null;
+            }
+            if (nextSibling instanceof DefinitionNode) {
+
```
Comment (NipunaMarcus) [inline tool-plugins/intellij/src/main/java/org/ballerinalang/plugins/idea/psi/references/DocVariableReference.java]: empty if ... can we remove this?

File: tool-plugins/intellij/src/main/java/org/ballerinalang/plugins/idea/psi/impl/BallerinaPsiImplUtil.java [hunk 4]
```diff
@@ -2872,4 +2872,28 @@ public static boolean isRedeclaredVar(@NotNull IdentifierPSINode identifier) {
         }
         return false;
     }
+
+    /**
+     * @param definitionNode
+     * @return
```
Comment (nadeeshaan) [inline tool-plugins/intellij/src/main/java/org/ballerinalang/plugins/idea/psi/impl/BallerinaPsiImplUtil.java]: Shall we add the return type description?

**Pull request merged**